### PR TITLE
Add access() to fs test and fix it on Windows

### PR DESCRIPTION
**Pull request by thomasten**
The access() function wasn't covered by the test yet.

### CODE REVIEW

**Review by mingweishih — APPROVED** [review]

**Review by anakrish — APPROVED** [review]

**Comment by anakrish** [comment]
bors try

**Comment by anakrish** [comment]
bors try

File: host/windows/syscall.c [hunk 1]
```diff
@@ -1883,10 +1884,6 @@ int oe_syscall_access_ocall(const char* pathname, int mode)
     {
         free(wpathname);
     }
-    if (pACL)
```
Comment (mingweishih) [inline host/windows/syscall.c:1886]: Why this removed?

Comment (thomasten) [reply host/windows/syscall.c:1886]: pACL is a non-owning pointer into the memory of pSD

Comment (mingweishih) [reply host/windows/syscall.c:1886]: Got it. Can you squash the commits? Thanks!

**Pull request merged**